### PR TITLE
ocamlPackages.progress: 0.1.1 → 0.2.1

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

#154901

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc maintainer @sternenseemann

### CODE REVIEW

File: pkgs/development/ocaml-modules/terminal/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/CraigFe/progress/releases/download/${version}/terminal-${version}.tbz";
+    sha256 = "sha256:0vjqkvmpyi8kvmb4vrx3f0994rph8i9pvlrz1dyi126vlb2zbrvs";
+  };
+
+  propagatedBuildInputs = [ stdlib-shims uutf uucp ];
+
+  doCheck = lib.versionAtLeast ocaml.version "4.05";
+  checkInputs = [ alcotest fmt ];
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/terminal/default.nix:21]: Also lists `astring` and `mtime` in opam file. Not sure if they are unneeded or propagated.

**Review by sternenseemann — APPROVED** [review]

**Comment by sternenseemann** [comment]
0.1.1 -> 0.2.1 is a breaking change, so *technically* we can't backport this, but the question is if anyone uses `progress` directly…

File: pkgs/development/ocaml-modules/terminal/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/terminal/default.nix:21]: The `opam` file is probably out of date: https://github.com/CraigFe/progress/blob/0.2.1/src/terminal/tests/dune

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/terminal/default.nix:21]: Checks out.

**Pull request merged**